# -*- coding: utf-8 -*-
#
# ./fix_apispec_cornice.diff
#
# This file is part of MUESLI.
#
# Copyright (C) 2018, Christian Heusel <christian (at) heusel.eu>
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.
--- helpers.py	2018-12-23 00:58:26.882353870 +0100
+++ helpers_fixed.py	2018-12-23 00:56:49.415471824 +0100
@@ -133,7 +133,8 @@
 
 
 def is_view(introspectable):
-    return introspectable.category_name == "views"
+    is_special_cornice = introspectable.title == 'function cornice.pyramidhook._fallback_view'
+    return introspectable.category_name == "views" and not is_special_cornice
 
 
 def check_methods_matching(view, **kwargs):
